<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PlannerAlgConfig</class>
 <widget class="QWidget" name="PlannerAlgConfig">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>360</width>
    <height>728</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Planning Algorithm Configuration</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QWidget" name="widget_2" native="true">
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="0" column="0">
       <widget class="QPushButton" name="load_button">
        <property name="text">
         <string>Load Configs...</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QPushButton" name="save_button">
        <property name="text">
         <string>Save Configs...</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="label_7">
     <property name="text">
      <string>Motion-Graph Storage Policy:</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QComboBox" name="graph_storage_selection">
     <item>
      <property name="text">
       <string>Adjacency-List Graph</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Adjacency-List / DVP-Tree Overlay</string>
      </property>
     </item>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="label_8">
     <property name="text">
      <string>Nearest-Neighbor Method:</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QComboBox" name="KNN_method_selection">
     <item>
      <property name="text">
       <string>Linear Search</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Binary Breadth-first DVP-Tree</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Quaternary Breadth-first DVP-Tree</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Binary Cache-Oblivious DVP-tree</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Quaternary Cache-Oblivious DVP-tree</string>
      </property>
     </item>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="label_5">
     <property name="text">
      <string>Motion Planning Method:</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QComboBox" name="planning_algo_selection">
     <item>
      <property name="text">
       <string>Rapidly-Exploring Random Tree (RRT)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Rapidly-Exploring Random Tree Star (RRT*)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Probabilistic Road-Map (PRM)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Sampling-based A* (SBA*)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Flexible Anytime-Dynamic PRM (FADPRM)</string>
      </property>
     </item>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="general_option_group">
     <property name="title">
      <string>General Planner Options</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>Maximum Number of Nodes:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>Progress-Report Interval:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string>Stop after:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSpinBox" name="maxvertices_spinbox">
        <property name="suffix">
         <string> nodes</string>
        </property>
        <property name="minimum">
         <number>100</number>
        </property>
        <property name="maximum">
         <number>20000</number>
        </property>
        <property name="singleStep">
         <number>100</number>
        </property>
        <property name="value">
         <number>2000</number>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QSpinBox" name="progress_interval_spinbox">
        <property name="suffix">
         <string> iterations</string>
        </property>
        <property name="minimum">
         <number>100</number>
        </property>
        <property name="maximum">
         <number>10000</number>
        </property>
        <property name="singleStep">
         <number>100</number>
        </property>
        <property name="value">
         <number>500</number>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QSpinBox" name="maxsolutions_spinbox">
        <property name="suffix">
         <string> solutions</string>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>200</number>
        </property>
        <property name="singleStep">
         <number>5</number>
        </property>
        <property name="value">
         <number>50</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Specific Planner Options</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QCheckBox" name="check_lazy_collision">
        <property name="text">
         <string>Lazy Collision Check</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_relaxation">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>        Init. Relaxation:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QCheckBox" name="check_anytime_heuristic">
        <property name="text">
         <string>Anytime Heuristic</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QDoubleSpinBox" name="init_sa_temp_spinbox">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="decimals">
         <number>1</number>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
        <property name="value">
         <double>2.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="label_start_delay">
        <property name="text">
         <string>Start Delay (s):</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_sa_temp">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>        Init. SA Temperature:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QCheckBox" name="check_voronoi_pull">
        <property name="text">
         <string>Add Voronoi Pull</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QCheckBox" name="check_bidir">
        <property name="text">
         <string>Bi-directional</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QCheckBox" name="check_bnb">
        <property name="text">
         <string>Use Branch-and-bound</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QDoubleSpinBox" name="init_relax_spinbox">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="decimals">
         <number>1</number>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
        <property name="value">
         <double>5.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="QDoubleSpinBox" name="start_delay_spinbox">
        <property name="suffix">
         <string> seconds</string>
        </property>
        <property name="decimals">
         <number>0</number>
        </property>
        <property name="maximum">
         <double>200.000000000000000</double>
        </property>
        <property name="value">
         <double>10.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="max_expansion_label">
        <property name="text">
         <string>Max Random-walk Distance:</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QDoubleSpinBox" name="max_expansion_spinbox">
        <property name="decimals">
         <number>1</number>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
        <property name="value">
         <double>1.000000000000000</double>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Progress Report Options</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QWidget" name="widget" native="true">
        <layout class="QGridLayout" name="gridLayout_3">
         <property name="margin">
          <number>0</number>
         </property>
         <item row="0" column="0">
          <widget class="QCheckBox" name="check_print_timing">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>Timing (terminal)</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QCheckBox" name="check_print_counter">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="text">
            <string>Node counter (terminal)</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QCheckBox" name="check_print_graph">
           <property name="text">
            <string>Motion-graph trace</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QCheckBox" name="check_print_best">
           <property name="text">
            <string>Best solution trace</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer_2">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>203</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
  <action name="actionUpdateAvailOptions">
   <property name="text">
    <string>UpdateAvailOptions</string>
   </property>
  </action>
  <action name="actionValuesChanged">
   <property name="text">
    <string>ValuesChanged</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>check_voronoi_pull</sender>
   <signal>toggled(bool)</signal>
   <receiver>init_sa_temp_spinbox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>94</x>
     <y>328</y>
    </hint>
    <hint type="destinationlabel">
     <x>265</x>
     <y>354</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>check_voronoi_pull</sender>
   <signal>toggled(bool)</signal>
   <receiver>label_sa_temp</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>94</x>
     <y>328</y>
    </hint>
    <hint type="destinationlabel">
     <x>94</x>
     <y>354</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>check_anytime_heuristic</sender>
   <signal>toggled(bool)</signal>
   <receiver>label_relaxation</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>94</x>
     <y>380</y>
    </hint>
    <hint type="destinationlabel">
     <x>94</x>
     <y>406</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>check_anytime_heuristic</sender>
   <signal>toggled(bool)</signal>
   <receiver>init_relax_spinbox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>94</x>
     <y>380</y>
    </hint>
    <hint type="destinationlabel">
     <x>265</x>
     <y>406</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>graph_storage_selection</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>actionValuesChanged</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>179</x>
     <y>70</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>KNN_method_selection</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>actionValuesChanged</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>179</x>
     <y>117</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>check_anytime_heuristic</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionValuesChanged</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>94</x>
     <y>416</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>check_bidir</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionValuesChanged</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>265</x>
     <y>339</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>check_bnb</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionValuesChanged</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>94</x>
     <y>468</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>check_lazy_collision</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionValuesChanged</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>94</x>
     <y>339</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>check_print_best</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionValuesChanged</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>265</x>
     <y>609</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>check_print_counter</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionValuesChanged</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>265</x>
     <y>584</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>check_print_graph</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionValuesChanged</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>94</x>
     <y>609</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>check_print_timing</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionValuesChanged</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>94</x>
     <y>584</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>check_voronoi_pull</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionValuesChanged</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>94</x>
     <y>364</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>init_relax_spinbox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>actionValuesChanged</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>265</x>
     <y>442</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>init_sa_temp_spinbox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>actionValuesChanged</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>265</x>
     <y>390</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>max_expansion_spinbox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>actionValuesChanged</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>265</x>
     <y>494</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>maxsolutions_spinbox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>actionValuesChanged</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>265</x>
     <y>276</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>maxvertices_spinbox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>actionValuesChanged</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>265</x>
     <y>222</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>planning_algo_selection</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>actionValuesChanged</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>179</x>
     <y>164</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>progress_interval_spinbox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>actionValuesChanged</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>265</x>
     <y>249</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>start_delay_spinbox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>actionValuesChanged</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>265</x>
     <y>521</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
